src/Processors/QtiV2/Out/AssessmentItemBuilder.php

```diff
@@ -35,6 +35,9 @@ public function build($itemIdentifier, $itemLabel, array $questions, $content =
     {
         // Initialise our <assessmentItem>
         $assessmentItem = new AssessmentItem($itemIdentifier, $itemIdentifier, false);
+        if (isset($questions[0])) {
```

Comment:
@rakeshqdsa I'm trying to understand the behavior around the `$itemLabel` here. I'm not generally comfortable with a method implicitly rewriting a required argument like this. I'm not sure how this method gets called at the moment, but I imagine it would be misleading if you pass an intentional `$itemLabel` only for it to be overridden without any clear control from the user's perspective.

Can you explain what the order of precedence should be? It seems to me that the only time the `$itemLabel ` parameter is now useful is when the item has no questions. If that is the case maybe it makes more sense to make `$itemLabel` optional, or change it to be some kind of `defaultItemLabel`.

Keen to hear your thoughts on this.

Reply:
Done 
Removed from here and passes from  parent function.